### DISCUSSION TITLE
It says compiling.... when doing any changes for fast refresh

### DISCUSSION THREAD

**Discussion by VivekNeel** [vercel/next.js#13078] [Help] [answered]
Our project has around 50 pages and multiple components. Doing any UI/Text changes to any component, fast refresh takes around 30 seconds. Each time i make changes, on the terminal i can see it's compiling... for so long and then refreshes the component. Any idea what's happening? 

I'm on `9.4.1-canary.5`

And here is my next.config.js 

```
// next.config.js
const withCSS = require('@zeit/next-css');
const withSourceMaps = require('@zeit/next-source-maps');
const withOptimizedImages = require('next-optimized-images');
const withBundleAnalyzer = require('@next/bundle-analyzer')({
  enabled: process.env.ANALYZE === 'true',
});
const withTM = require('next-transpile-modules')(['common']);
// pass the modules you would like to see transpiled
module.exports = withBundleAnalyzer(
  withTM(withCSS(
    withOptimizedImages(
      withSourceMaps({
        experimental: { granular_chunks: false },
        compress: true,
        devIndicators: {
          autoPrerender: true,
        },
        assetPrefix: process.env.ASSETPREFIX
          ? `https://${process.env.ASSETPREFIX}.s3.amazonaws.com`
          : '', 
        webpack: config => config,
      // env: {
      //   STRIPE_KEY: process.env.STRIPE_KEY,
      // },
      }),
    ),
  )),
);

```

**Comment by VivekNeel** [comment]
babel.config.js 

```
module.exports = {
  presets: ['next/babel'],
  plugins: [
    'relay',
    'transform-flow-strip-types',
    [
      '@babel/plugin-syntax-dynamic-import',
      {
        libraryName: '@material-ui/core',
        libraryDirectory: 'esm',
        camel2DashComponentName: false,
      },
      'core',
    ],
    [
      '@babel/plugin-syntax-dynamic-import',
      {
        libraryName: '@material-ui/icons',
        libraryDirectory: 'esm',
        camel2DashComponentName: false,
      },
      'icons',
    ],
  ],
};

```

**Comment by VivekNeel** [comment]
can anyone help please?

**Comment by klaasman** [comment]
I'm also experiencing hot updates (or fast refreshes) taking 30+ seconds with next@9.4.0/9.4.1. Not really sure what could cause this, but maybe someone can give some handles in what direction to debug.

**Reply by VivekNeel** [reply]
Did you find any solution? For me it roughly takes around 20+ seconds :(

**Comment by Timer** [comment, marked as answer]
Could you please try `next@9.4.2-canary.4` or newer?

**Reply by VivekNeel** [reply]
Thank you Timer. Will try it and let you know. I was on next@9.4.2-canary.3

**Reply by Timer** [reply]
9.4.2 canary `.4` should've included a fix for this, that was not present in `.3`.

**Reply by klaasman** [reply]
Seems to work for me 👍 thanks

**Reply by VivekNeel** [reply]
Yup, now it just takes ~6 seconds which is a lot better than 20+ seconds for a larger project.
Hopefully in future it will get reduced to less than 2 s. Thank you for the fix

**Reply by Timer** [reply]
What kind of hardware are you on?

**Reply by VivekNeel** [reply]
![image](https://user-images.githubusercontent.com/5962275/82639736-eef5d500-9c26-11ea-8306-5eac7a43d15d.png)


Also i noticed that when i check activityMonitor, the app consumes around 1.5 GB for each page 🤔
